### PR TITLE
PSR-2 standard fully adopted by the project

### PR DESCRIPTION
**Pull request by luizperes**
This PR is related to the issue [#9](https://github.com/quack/quack/issues/9).

What does it have?
- [x] Applies PSR-2 to **all PHP files** under `src` and `test`
- [x] Fixes **all** PSR-2 `MUSTs`
- [x] Adds a `pre-commit` hook under the folder `/path/to/quack/tools/phpcs-pre-commit`
- [x] Changes the `CONTRIBUTING.md` file.
- [x] makes the use of `PHPCodeSniffer` and `PSR-2 coding style standard` mandatory to all of us.

What is still missing?
- We still need to **fix** all PSR-2 `SHOULDs`. (After installing `phpcs`, run the command  `phpcs --standard=PSR2 /path/to/quack`
- Code refactoring for all PSR-2 `SHOULDs`.

Obs.: The pre-commit only prevents you to commit if you do not follow PSR-2 `MUSTs` (a.k.a. errors). For warnings(a.k.a. SHOULDs), it will allow you to commit normally.

### CODE REVIEW

**Comment by joaolucasl** [comment]
Regardless of the things I pointed in #9, I think all [files should have a blank line in the end](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#22-files). Would you mind adding that to your changes? :) 
Thanks again for the PR, and sorry for the nitpicking!

**Comment by haskellcamargo** [comment]
These additions come before your forked the project:
- [https://github.com/quack/quack/pull/33](Short syntax for open and expression disambiguation on arrays)
- [https://github.com/quack/quack/pull/32](Minor updates in QuackRepl)

GitHub doesn't say it's a merge conflict, although. However, we can still look at the _diff_ and do the merging by hand.

**Comment by haskellcamargo** [comment]
I reviewed it. The methods were reordered all the **MUST**s were fixed. I rechecked and what I mentioned before didn't cause any break, because you already did a `git pull` before!

Definitely, LGTM!

**Comment by haskellcamargo** [comment]
Ah, I would only ask, if possible, to merge [these changes](https://github.com/quack/quack/pull/33/files) in your PR. They are syntactic extensions for `open` statement that were later added; also, they solve an ambiguation in assocative arrays.

**Comment by luizperes** [comment]
@joaolucasl I thought I had done that already! Good point! (regarding LF at the end).

@haskellcamargo I will merge them in my PR! :)

**Comment by luizperes** [comment]
I think that the merge is done :) Could anyone check it please?

File: TODO.md [hunk 1]
```diff
@@ -20,12 +20,8 @@
 - [x] Implement `<expr> .. <expr> by <expr>` syntax
 - [x] Implement atoms as expressions
 - [x] Create a single parselet for literal values
-- [x] Verify associative array definitions
-- [x] Implement short syntax for `open` statements
+- [ ] Implement short syntax for `open` statements
 - [ ] Implement partial function parsing
 - [ ] Implement static properties (without semantic validation)
 - [ ] Verify property definitions for classes
-- [ ] Make print an expression and stdout an alias for echo
```
Comment (haskellcamargo) [inline TODO.md:28]: Why have these sections been removed from the file? The `TODO.md` should be kept as the current in the repository.

File: bootstrap/parser/Parser.qk [hunk 2]
```diff
@@ -2,8 +2,8 @@
 module QuackCompiler.Parser
 
 open .Exception
-open .QuackCompiler.Lexer { Tag; Token; Tokenizer }
-open .QuackCompiler.Parselets {
+open QuackCompiler.Lexer.{ Tag; Token; Tokenizer }
+open QuackCompiler.Parselets.{
```
Comment (haskellcamargo) [inline bootstrap/parser/Parser.qk]: This file too should be kept as the repository version.

**Comment by haskellcamargo** [comment]
I've pointed out what looks unfixed. Those I commented in your last commit were the only issues I found. The file `TODO.md` and the Quack self-hosted initial parser should be like the origin version.

**Comment by luizperes** [comment]
Oops..my bad, i must've replaced them with one my regex or Atom did it itself. I'll fix it. Was it in this last commit?

**Comment by haskellcamargo** [comment]
Yes, but only these two files! I believe that, after doing that, you can give a self LGTM, as much as I already reviewed, and we can merge!

**Comment by haskellcamargo** [comment]
I just forked your repository and added the missing files. I've made a [pull-request](https://github.com/luizperes/quack/pull/1) there. If you merge it, we only need somebody (or yourself) to give us a _LGTM_ in order to merge. I've tested your changes locally and didn't see any unexpected behavior. @sergiors, after the sub-merge, can you assert it and apply a _LGTM_, please?

**Comment by haskellcamargo** [comment]
Nice! The tests were expected to break because I did changes in the lexer. I'll fix them and resend you another pull-request! @luizperes

**Comment by haskellcamargo** [comment]
Done [here](https://github.com/luizperes/quack/pull/2). Merge it and comment "LGTM" in order to allow us merge.

**Comment by sergiors** [comment]
Sorry, I am away from computer 😐

**Comment by haskellcamargo** [comment]
@luizperes, the LGTM should be in this repository.

**Comment by luizperes** [comment]
LGTM

**Pull request merged**